lib/facter/consul_version.rb

```diff
@@ -0,0 +1,17 @@
+# consul_version.rb
+
+Facter.add(:consul_version) do
+  confine :kernel => 'Linux'
+  setcode do
+    Facter::Core::Execution.exec('consul --version | head -1 | cut -f2 -d\ | cut -c2-')
```

Comment:
That's a very good idea, can't believe I didn't think of it rather!  Will get back to this.